src/main/java/core/basesyntax/strategy/impl/PurchaseTransaction.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.StorageTransaction;
+import core.basesyntax.strategy.Transaction;
+
+public class PurchaseTransaction implements Transaction {
+    private static final int ZERO_QUANTITY = 0;
+
+    @Override
+    public void doTransaction(StorageTransaction transaction) {
+        if (Storage.STORAGE.get(transaction.getFruit()) >= transaction.getQuantity()) {
+            Storage.STORAGE.merge(transaction.getFruit(), -transaction.getQuantity(), Integer::sum);
+        } else {
+            Storage.STORAGE.put(transaction.getFruit(), ZERO_QUANTITY);
```

Comment:
Why did you set zero amount? You should throw exceptions with an understandable message. For example, if you have 100 bananas, I want to buy 101, what might happen?